src/components/Checkbox/index.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react'
+import styled, { css } from 'styled-components'
+import { theme } from '../../theme'
+import { Label } from '../Label'
+
+const CheckboxContainer = styled.div`
+  display: inline-block;
+  vertical-align: middle;
+`
+
+const HiddenCheckbox = styled.input.attrs({ type: 'checkbox' })`
+  border: 0;
+  clip: rect(0 0 0 0);
+  clippath: inset(50%);
+  height: 1px;
+  margin: -1px;
+  overflow: hidden;
+  padding: 0;
+  position: absolute;
+  white-space: nowrap;
+  width: 1px;
+`
+
+const Icon = styled.svg`
+  fill: none;
+  stroke: ${theme.fullWhite};
+  stroke-width: 2px;
+`
+
+const StyledCheckbox = styled.div<{ checked?: boolean, $disabled?: boolean }>`
+  display: inline-block;
+  border: 1px solid ${props => props.checked ? theme.primary : theme.grey50};;
+  margin-right: 10px;
+  width: 16px;
+  height: 16px;
+  background: ${props => props.checked ? theme.primary : 'transparent'};
+  display: flex;
+  transition: all 150ms;
+  ${HiddenCheckbox}:focus + & {
+    box-shadow: 0 0 0 3px ${theme.grey40};
+  }
+  ${({$disabled}) => $disabled ? css`
+    pointer-events: none;
+    background-color: ${theme.grey10};
+    border-color: ${theme.grey30};
+    svg {
+      stroke: ${theme.grey40};
+    }
+  ` : ''};
+`
+
+const HorizontalLabel = styled(Label)`
+  display: flex;
+  margin-bottom: 0px;
+  align-items: center;
+`
+
+export type CheckboxProps = {
+  checked?: boolean
```

Comment:
we should make this required if the component only works as a [controlled component](https://fb.me/react-controlled-components) or we should fix the behavior when it is uncontrolled.

right now it doesn't work when uncontrolled, i.e. if `checked` is not passed in, the underlying checkbox in `HiddenCheckbox` will change its internal state when clicked, but the appearance of the `StyledCheckbox` does not change. you can see this with storybook:
1. launch storybook http://localhost:6006/?path=/story/common-checkbox--primary
2. click on the Actions tab
3. click on the checkbox repeatedly
4. observe the appearance doesn't change but the onChange handler logs alternating values